File: 2305-append-k-integers-with-minimal-sum/append-k-integers-with-minimal-sum.java
class Solution {
    public long minimalKSum(int[] nums, int k) {
        Arrays.sort(nums);
        long sum = 0;
        int prev = 0;
        for(int num : nums){
            if(num - prev > 1 && k > 0){
                long st = prev + 1;
                long en = Math.min(num-1, st+k-1);
                long cnt = en - st + 1;
                sum += (st + en) * cnt/2;
                k -= cnt;
            }
            prev = num;
            if(k == 0) break;
        }
        if(k > 0){
            long st = prev+1;
            long en = st + k - 1;
            sum += (st + en) * k/2;
        }
        return sum;
    }
}

File: 2305-append-k-integers-with-minimal-sum/append-k-integers-with-minimal-sum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minimalKSum = function (nums, k) {
    nums.sort((a, b) => a - b);
    let sum = 0n;  // Use BigInt for large sums
    let prev = 0;

    for (let num of nums) {
        if (num - prev > 1 && k > 0) {
            let start = BigInt(prev + 1);
            let end = BigInt(Math.min(num - 1, prev + k));
            let count = end - start + 1n;
            sum += (start + end) * count / 2n;
            k -= Number(count);
        }
        prev = num;
        if (k === 0) break;
    }

    if (k > 0) {
        let start = BigInt(prev + 1);
        let end = BigInt(start + BigInt(k) - 1n);
        sum += (start + end) * BigInt(k) / 2n;
    }

    return Number(sum);
};